iwyu.cc

```diff
@@ -1768,6 +1770,11 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
         while ((redecl = redecl->getPreviousDecl()))
           ReportDeclUse(CurrentLoc(), redecl);
       }
+      if (!IsInHeader(decl)) {
```

Comment:
I just ran into clang's `SourceManager::isMainFile(const FileEntry&)`, which I think might be an even better fit.

Unless I'm missing something, we only ever have one `.cpp` file in flight -- the main compilation unit. So anywhere `isMainFile` returns false, we're in some other file, and that must be a header (or some eccentric thing like included .cpp file).

I would generally prefer that to adding semantic weight to filenames, so please give it a shot and see if it works. There's a `GlobalSourceManager()` accessor available for the source manager.